src/framework/server.ts

```diff
@@ -1,51 +1,21 @@
-import Koa from 'koa';
-import compose from 'koa-compose';
-import {
-  ErrorMiddleware,
-  MetricsMiddleware,
-  RequestLogging,
-  // SecureHeaders,
-  VersionMiddleware,
-} from 'seek-koala';
+import { FastifyPluginAsync, FastifyPluginCallback, fastify } from 'fastify';
 
-import { config } from 'src/config';
-import { contextMiddleware, logger } from 'src/framework/logging';
-import { metricsClient } from 'src/framework/metrics';
+import { contextHook } from './context';
+import { errorHandler } from './error';
+import { errorLoggingHook, responseLoggingHook, versionHook } from './hooks';
 
-const metrics = MetricsMiddleware.create(
-  metricsClient,
-  ({ _matchedRoute }) => ({
-    route: typeof _matchedRoute === 'string' ? _matchedRoute : 'unspecified',
-  }),
-);
+export const createApp = async (
+  ...plugins: (FastifyPluginAsync | FastifyPluginCallback)[]
+) => {
+  const server = fastify();
+  server.setErrorHandler(errorHandler);
 
-const requestLogging = RequestLogging.createMiddleware((ctx, fields, err) => {
-  if (ctx.status < 400 && err === undefined) {
-    // Depend on sidecar logging for happy path requests
-    return;
-  }
+  server.addHook('onRequest', contextHook);
+  server.addHook('onError', errorLoggingHook);
```

Comment:
Likely will convert these to plugins in a seek-fastify library. For ease of use right now I'll leave them as hooks